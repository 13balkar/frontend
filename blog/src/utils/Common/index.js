import monthNames from '../../constants/monthNames';
const getDaySuffix = (day) => {
  if (day > 3 && day < 21) return 'th';
  switch (day % 10) {
    case 1: return 'st';
    case 2: return 'nd';
    case 3: return 'rd';
    default: return 'th';
  }
};

const getFormattedDate = (date) => {
  date = new Date(date);
  const day = date.getDate();
  const month = date.getMonth();
  const year = date.getFullYear();
  return `${day}${getDaySuffix(day)} ${monthNames[month]}/${year}`;
};

export default getFormattedDate;
