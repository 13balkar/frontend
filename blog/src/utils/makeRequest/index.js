import axios from 'axios';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig) => {
  const requestConfig = {
    url: `${BACKEND_URL}${apiEndPoint.url}`,
    method: apiEndPoint.method,
    ...dynamicConfig
  };
  const { data } = await axios(requestConfig);
  return data;
};

export default makeRequest;
