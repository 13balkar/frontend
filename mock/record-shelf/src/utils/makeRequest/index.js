import axios from 'axios';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig) => {
  const requestConfig = {
    url: `${BACKEND_URL}${apiEndPoint.url}`,
    headers: {
      authorization: 'Bearer QWlzaHdhcnlhIE4'
    },
    method: apiEndPoint.method,
    ...dynamicConfig
  };
  const { data } = await axios(requestConfig);
  return data.data;
};

export default makeRequest;
