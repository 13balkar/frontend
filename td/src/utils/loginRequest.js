import axios from 'axios';

const register = async (dynamicConfig, navigate) => {
  try {
    const requestConfig = {
      url: 'http://localhost:4000/user/create',
      method: 'post',
      ...dynamicConfig
    };
    const { data } = await axios(requestConfig);
    return data;
  } catch (error) {
    if (navigate) {
      const errorStatus = error.response?.status;
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};
const login = async (dynamicConfig, navigate) => {
  console.log('dynamicConfig', dynamicConfig);
  try {
    const requestConfig = {
      url: 'http://localhost:4000/user/login',
      method: 'post',
      ...dynamicConfig
    };
    const { data } = await axios(requestConfig);
    localStorage.setItem('token', data.token);
    localStorage.setItem('username', dynamicConfig.data.userName);
    navigate('/todo');
    // return data;
  } catch (error) {
    if (navigate) {
      const errorStatus = error.response?.status;
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};

export { register, login };
