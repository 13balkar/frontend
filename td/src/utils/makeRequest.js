import axios from 'axios';
import { BACKEND_URL } from '../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig, navigate) => {
  try {
    const requestConfig = {
      url: `${BACKEND_URL}${apiEndPoint.url}`,
      method: apiEndPoint.method,
      ...dynamicConfig
    };
    // console.log('requestConfig', requestConfig);
    const { data } = await axios(requestConfig);
    return data;
  } catch (error) {
    if (navigate) {
      const errorStatus = error.response?.status;
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};

export default makeRequest;
