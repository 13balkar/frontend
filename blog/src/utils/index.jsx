import getFormattedDate from './Common';
import makeRequest from './makeRequest';

export { getFormattedDate, makeRequest };
