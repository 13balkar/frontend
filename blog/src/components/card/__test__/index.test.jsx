import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import * as React from 'react';
import Cards from '..';
import { makeRequest } from '../../../utils';
import Card from '../card';
import Reaction from '../reactions';
jest.mock('../../../utils/makeRequest');

const mockData = {
  id: 2,
  date: '6/29/2022 4:52:48 PM UTC',
  reading_time: '4 mins',
  title: 'mock title 2',
  description: 'mock description 2',
  claps: 12,
  liked: true,
  image: 'abstract.png'
};

describe('Data Fetching', () => {
  it('should show loading when the data is being fetched', async () => {
    makeRequest.mockResolvedValue([mockData]);
    render(<Cards />);
    await waitFor(() =>
      expect(screen.getByText('Loading')).toBeTruthy()
    );
  });
  it('should show the blog posts when the data is fetched', async () => {
    makeRequest.mockResolvedValue([mockData]);

    render(<Cards />);
    await waitFor(() =>
      expect(screen.getAllByText('mock title 2')).toBeTruthy()
    );
  });
  it('should show all the blog posts when the data is fetched', async () => {
    makeRequest.mockResolvedValue([mockData]);
    render(<Cards />);
    await waitFor(() =>
      expect(screen.getAllByTestId('blog')).toHaveLength([mockData].length)
    );
  });
  it('should show error when there is error in data fetching', async () => {
    makeRequest.mockRejectedValue({ message: 'Error' });
    render(<Cards />);
    await waitFor(() =>
      expect(screen.getAllByText('Error')).toBeTruthy()
    );
  });
});

describe('Card Component', () => {
  it('should display the the given count of claps when rendered for the first time', () => {
    makeRequest.mockResolvedValueOnce(mockData);
    const { getByText } = render(<Card post={mockData} />);
    expect(getByText(mockData.claps)).toBeTruthy();
  });
  it('should increment the number count of claps when clicked on clap icon', async () => {
    makeRequest.mockResolvedValueOnce(mockData);
    const { getByText } = render(<Card post={mockData} />);
    expect(getByText(mockData.claps)).toBeTruthy();
    const clapIcon = getByText(mockData.claps).parentElement.querySelector('img');
    await waitFor(() => {
      fireEvent.click(clapIcon);
      expect(getByText(mockData.claps + 1)).toBeTruthy();
    });
  });
  it('should display the given color of heart when rendered for the first time', () => {
    makeRequest.mockResolvedValueOnce(mockData);
    const { getByAltText } = render(<Card post={mockData} />);
    expect(getByAltText('like')).toHaveAttribute('src', '/assets/Icons/heart-red.svg');
  });
  it('should change the color of the color of heart icon when clicked on heart icon', async () => {
    const { getByAltText } = render(<Card post={mockData} />);

    expect(getByAltText('like')).toHaveAttribute('src', '/assets/Icons/heart-red.svg');
    const heartIcon = getByAltText('like');
    await waitFor(() => {
      fireEvent.click(heartIcon);
      expect(getByAltText('like')).toHaveAttribute('src', '/assets/Icons/heart-black.svg');
    });
  });
  it('should call the increment claps function when clicked on clap icon', async () => {
    const clapping = jest.fn();
    const { getByText } = render(<Reaction count={mockData.claps} clapping={clapping} liked={mockData.liked} liker={jest.fn()} />);
    const clapIcon = getByText(mockData.claps).parentElement.querySelector('img');
    await waitFor(() => {
      fireEvent.click(clapIcon);
      expect(clapping).toHaveBeenCalled();
    });
  });
  it('should call the liker function when clicked on heart icon', async () => {
    const liker = jest.fn();
    const { getByAltText } = render(<Reaction count={mockData.claps} liked={mockData.liked} clapping={jest.fn()} liker={liker} />);
    const heartIcon = getByAltText('like');
    await waitFor(() => {
      fireEvent.click(heartIcon);
      expect(liker).toHaveBeenCalled();
    });
  });
});

describe('Snapshot', () => {
  it('should render card list correctly', async () => {
    makeRequest.mockResolvedValue([mockData]);
    const { asFragment } = render(<Cards />);
    await waitFor(() =>
      expect(asFragment()).toMatchSnapshot()
    );
  });
  it('should render card correctly', () => {
    const { asFragment } = render(<Card post={mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
