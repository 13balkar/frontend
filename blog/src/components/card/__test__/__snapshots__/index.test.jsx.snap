// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should render card correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="abcd"
      class="container-img"
      src="abstract.png"
    />
    <div
      class="container-data"
    >
      <div
        class="container-feat"
      >
        <p>
          29th June/2022
        </p>
        <p>
          4 mins
        </p>
      </div>
      <h3>
        mock title 2
      </h3>
      <p>
        mock description 2
      </p>
      <hr />
    </div>
    <div
      class="container-data"
    >
      <div
        class="container-feat"
      >
        <div
          class="reactions"
        >
          <img
            alt=""
            src="/assets/Icons/clapping.svg"
          />
          <p>
            12
          </p>
        </div>
        <img
          alt="like"
          src="/assets/Icons/heart-red.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot should render card list correctly 1`] = `
<DocumentFragment>
  <h1>
    Loading
  </h1>
</DocumentFragment>
`;
